setuptools/msvc.py

```diff
@@ -212,6 +221,19 @@ def msvc14_get_vc_env(plat_spec):
         raise
 
 
+def msvc14_gen_lib_options(*args, **kwargs):
+    """
+    Patched "distutils._msvccompiler.gen_lib_options" for fix 
+    compatibility between "numpy.distutils" and "distutils._msvccompiler"
+    (for Numpy < 1.11.2)
+    """
+    if "numpy.distutils" in sys.modules:
+        import numpy as np
+        if StrictVersion(np.__version__) < StrictVersion('1.11.2'):
```

Comment:
The problem with parse version is it will return a LegacyVersion when a proper Version can't be constructed and a LegacyVersion always compares less than a Version.

The reason for extern.packaging is because Setuptools vendors the packaging package, and extern exposes the package whether it is vendored or not.